include/retdec/fileformat/file_format/pe/pe_format_parser.h

```diff
@@ -0,0 +1,656 @@
+										remainder);
+	}
+
+	std::uint32_t getFileAlignment() const
+	{
+		return peFile->imageLoader().getFileAlignment();
+	}
+		
+	std::uint32_t getSectionAlignment() const
+	{
+		return peFile->imageLoader().getSectionAlignment();
+	}
+
+	std::uint32_t getSizeOfHeaders() const
+	{
+		return peFile->imageLoader().getSizeOfHeaders();
+	}
+
+	std::uint32_t getSizeOfImage() const
+	{
+		return peFile->imageLoader().getSizeOfImage();
+	}
+
+	std::uint32_t getChecksum() const
+	{
+		return peFile->imageLoader().getOptionalHeader().CheckSum;
+	}
+
+	std::uint64_t getSizeOfStackReserve() const
+	{
+		return peFile->imageLoader().getOptionalHeader().SizeOfStackReserve;
+	}
+
+	std::uint64_t getSizeOfStackCommit() const
+	{
+		return peFile->imageLoader().getOptionalHeader().SizeOfStackCommit;
+	}
+
+	std::uint64_t getSizeOfHeapReserve() const
+	{
+		return peFile->imageLoader().getOptionalHeader().SizeOfHeapReserve;
+	}
+
+	std::uint64_t getSizeOfHeapCommit() const
+	{
+		return peFile->imageLoader().getOptionalHeader().SizeOfHeapCommit;
+	}
+
+	std::uint32_t getSizeOfPeSignature() const
+	{
+		return sizeof(std::uint32_t);
+	}
+
+	std::uint32_t getLoadedSizeOfNtHeaders() const
+	{
+		return peFile->imageLoader().getFileHeader().SizeOfOptionalHeader;
+	}
+
+	std::uint32_t getAllocatedSizeOfNtHeaders() const
+	{
+		return peFile->imageLoader().getFileHeader().SizeOfOptionalHeader;
+	}
+
+	std::uint32_t getDeclaredNumberOfDataDirectories() const
+	{
+		return peFile->imageLoader().getOptionalHeader().NumberOfRvaAndSizes;
+	}
+
+	std::uint32_t getStoredNumberOfDataDirectories() const
+	{
+		return peFile->imageLoader().getRealNumberOfDataDirectories();
+	}
+
+	std::uint32_t getNumberOfImportedLibraries() const
+	{
+		return peFile->impDir().getNumberOfFiles(PeLib::OLDDIR);
+	}
+
+	std::uint32_t getNumberOfDelayImportedLibraries() const
+	{
+		return peFile->delayImports().getNumberOfFiles();
+	}
+
+	bool isDll() const
+	{
+		return (peFile->imageLoader().getCharacteristics() & PeLib::PELIB_IMAGE_FILE_DLL);
+	}
+
+	bool getEpAddress(std::uint64_t & epAddress) const
+	{
+		std::uint64_t imageBase = peFile->imageLoader().getImageBase();
+		std::uint32_t entryPoint = peFile->imageLoader().getOptionalHeader().AddressOfEntryPoint;
+
+		// Do not report zero entry point on DLLs
+		epAddress = imageBase + entryPoint;
+		return (entryPoint != 0 || isDll() == false);
+	}
+
+	bool getEpOffset(std::uint64_t & epOffset) const
+	{
+		std::uint32_t entryPoint = peFile->imageLoader().getOptionalHeader().AddressOfEntryPoint;
+
+		epOffset = peFile->imageLoader().getFileOffsetFromRva(entryPoint);
+		return (entryPoint != 0 || isDll() == false) && (epOffset != UINT32_MAX);
+	}
+
+	bool getSectionType(const PeLib::PELIB_SECTION_HEADER * pSectionHeader, PeCoffSection::Type & secType) const
+	{
+		std::uint32_t Characteristics = pSectionHeader->Characteristics;
+
+		if(Characteristics & (PeLib::PELIB_IMAGE_SCN_CNT_CODE | PeLib::PELIB_IMAGE_SCN_MEM_EXECUTE))
```

Comment:
Yes, this is the same code. If the original was encloses in parentheses better, it would be:
`if((flags & PeLib::PELIB_IMAGE_SCN_CNT_CODE) || (flags & PeLib::PELIB_IMAGE_SCN_MEM_EXECUTE))`
aka check if either `PELIB_IMAGE_SCN_CNT_CODE` or `PELIB_IMAGE_SCN_MEM_EXECUTE` is set.